### PR TITLE
dts: arm: st: Remove use of CONFIG_SOC_* from STM32 F2/F3/F4/F6 dts files

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: dts/arm/st/f4/stm32f417Xe.dtsi [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <st/f4/stm32f405.dtsi>
```
Comment (erwango) [inline dts/arm/st/f4/stm32f417Xe.dtsi]: 407?

File: dts/arm/st/f4/stm32f417Xg.dtsi [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <st/f4/stm32f405.dtsi>
```
Comment (erwango) [inline dts/arm/st/f4/stm32f417Xg.dtsi]: 407?

**Review by erwango — CHANGES_REQUESTED** [review]

File: dts/arm/st/f4/stm32f417Xg.dtsi (hunk 2 above)
Comment (ydamigos) [reply dts/arm/st/f4/stm32f417Xg.dtsi]: 417?

File: dts/arm/st/f4/stm32f417Xe.dtsi (hunk 1 above)
Comment (ydamigos) [reply dts/arm/st/f4/stm32f417Xe.dtsi]: 417?

**Review by ydamigos — CHANGES_REQUESTED** [review]

File: dts/arm/st/f4/stm32f417Xe.dtsi (hunk 1 above)
Comment (galak) [reply dts/arm/st/f4/stm32f417Xe.dtsi]: Fixed

File: dts/arm/st/f4/stm32f417Xg.dtsi (hunk 2 above)
Comment (galak) [reply dts/arm/st/f4/stm32f417Xg.dtsi]: Fixed, made it 417.

**Review by ydamigos — APPROVED** [review]

**Pull request merged**